.github/scripts/make_mocs.py

```diff
@@ -0,0 +1,233 @@
+            link_name += extension
+        return self.make_link_line(directory, link_name)
+
+    def make_line_for_sub_directory(self, directory, sub_directory):
+        path = directory + '/' + sub_directory
+        namer = MocFileNamer()
+        file = namer.moc_name_for_sub_directory(sub_directory)
+        return self.make_link_line(path, file)
+
+    def strip_parent_directories_from_directory(self, directory):
+        # Ugly hack because all directory names start with '../../'
+        result = directory.replace('../', '')
+        if result == '..':  # Even more horrible hack for files in root of repo
+            result = ''
+        return result
+
+    def make_link_line(self, directory, link_name):
+        adjusted_directory = self.strip_parent_directories_from_directory(directory)
+        if len(adjusted_directory) > 0:
+            adjusted_directory += '/'
+        result = f'-  [[{adjusted_directory}{link_name}|{link_name}]]\n'
+        # print(f'directory={directory}\nadjusted_directory={adjusted_directory}\nlink_name={link_name}\n=> {result}')
+        return result
+
+
+class MocFileAndDirectoryFilter:
+    """Various filtering functions, to determine what is included in the generated MOC"""
+
+    def __init__(self):
+        self.DIRECTORIES_TO_EXCLUDE = ['meta-notes', 'venv']  # Directories beginning '.' are also excluded
+
+    def include_directory_in_moc(self, directory):
+        if directory[0] == '.':
+            return False
+        return directory not in self.DIRECTORIES_TO_EXCLUDE
+
+    def include_file_in_moc(self, directory, file):
+        if self.file_is_moc_for_directory(directory, file):
+            return False
+        return file[0] != '.'
+
+    def file_is_moc_for_directory(self, directory, file):
+        namer = MocFileNamer()
+        return file == namer.moc_file_name_for_directory(directory)
+
+    def filter_directories(self, dirs):
+        dirs[:] = [d for d in dirs if self.include_directory_in_moc(d)]
+
+
+class MocFileNamer:
+    """
+    Various naming functions, to determine the name and location of MOC files.
+
+    Currently, it is hard-coded that the Moc file for a directory
+    will be called the directory name, with a 🗂️ prefix.
+
+    This matches the configuration of the Zoottelkeeper plugin as of November 2021.
+    """
+
+    def moc_name_for_sub_directory(self, sub_directory):
+        name = sub_directory
+        if name == '..':
+            name = 'hub'
+        return '🗂️ ' + name
+
+    def moc_base_name_for_directory(self, root):
+        directory_name = os.path.basename(root)
+        return self.moc_name_for_sub_directory(directory_name)
+
+    def moc_file_name_for_directory(self, root):
+        return self.moc_base_name_for_directory(root) + ".md"
+
+    def moc_file_path_for_directory(self, root):
+        moc_file_basename = self.moc_file_name_for_directory(root)
+        return os.path.join(root, moc_file_basename)
+
+
+class MocDelimiter:
```

Comment:
These could be constants, rather than methods on the class.

Reply:
Are you meaning the 4 methods in `MocFileNamer`?

This is one of the examples where the argument name `root` is misleading - these are called multiple times, as the directory tree is walked...

I haven't been able to come up with a more descriptive variable name - partly because I haven't been able to run the code in a debugger in PyCharm on my Mac, making it harder to inspect the data during execution.

Reply:
I've renamed root to directory, to try to make it clearer that it's not a constant.